### ISSUE TITLE
DAT lack SHA1digests

### ISSUE THREAD

**Issue by majandres** [andrebrait/1g1r-romset-generator#21]
I'm using the FBneo dat (arcade only) file found [here ](https://github.com/libretro/FBNeo/tree/master/dats)

and when I try to create a 1g1r set, I get this error:
`ERROR: Cannot use hash information because DAT lacks SHA1 digests for [ym2608]`

Looks like the DAT file only contains CRC and no SHA1 or MD5 hashes.

Is CRC usage enough? or will SHA1 be needed in the DAT files?

Thank you, this is a really great tool you built here.

**Comment by andrebrait** [comment]
I can potentially use CRC or even MD5, but these are less-than-ideal methods. Some major refactoring is in the works, so that might come :)

**Comment by majandres** [comment]
appreciate the response.  thank you.

**Comment by prenetic** [comment]
A few of the No-Intro DATs include `[MIA]` entries which also fall into this category -- only CRC, no MD5 or SHA1. Would a viable option be to fall back to MD5 and ultimately CRC?. As it stands I've been adding a bogus SHA1 hash for dozens of these entries so the tool can function in scan mode with these DATs.

Example:
`[MIA] Arkanoid + Rampage (Europe) (Summer Pack)`

**Comment by andrebrait** [comment]
> A few of the No-Intro DATs include `[MIA]` entries which also fall into this category -- only CRC, no MD5 or SHA1. Would a viable option be to fall back to MD5 and ultimately CRC?. As it stands I've been adding a bogus SHA1 hash for dozens of these entries so the tool can function in scan mode with these DATs.
> 
> Example:
> `[MIA] Arkanoid + Rampage (Europe) (Summer Pack)`

Yeah, I started noticing these a while ago, but I'm not updating this tool anymore. There's one feature that I need to create a release for but I don't plan on refining it further.

I think https://github.com/andrebrait/DATROMTool already does this. It scans the files and adds filters them for SHA-1, MD5 and CRC+Size, in this order of priority.

I still have to finish a few things in there, but if you'd like you can grab the current pre-release build and give it a try and see if it works for your case

**Comment by prenetic** [comment]
Didn't realize there was a successor, I'll check that out as well -- thank you!

**Comment by andrebrait** [comment]
It isn't ready at the moment, but it's capable of doing everything this tool does, minus some occasional bug
